lib/grpc/client/adapters/mint/connection_process/connection_process.ex

```diff
@@ -172,6 +187,34 @@ defmodule GRPC.Client.Adapters.Mint.ConnectionProcess do
   end
 
   @impl true
+  def handle_info(:ping, %{ping_ref: nil} = state) do
+    case Mint.HTTP2.ping(state.conn) do
+      {:ok, conn, ref} ->
+        Process.send_after(self(), :ping, @keep_alive_timeout)
+
+        state =
+          state
+          |> State.update_conn(conn)
+          |> State.set_ping_ref(ref)
+
+        {:noreply, state}
+
+      {:error, conn, _error} ->
+        state = State.update_conn(state, conn)
+        check_connection_status(state)
+        {:noreply, state}
```

Comment:
You want the `check_connection_status` to handle the GS reply for you
```suggestion
        check_connection_status(state)
```